### PR TITLE
Return true when calling discard on discarded record

### LINKED ISSUE #20: Documentation for un/discard is not completely correct
https://github.com/jhawthorn/discard/blob/cb3c885849ce494b426a1e99d2049eaba8145f16/lib/discard/model.rb#L71-L75

I think either these should return `false` if it's already discarded or the documentation should read "true if successful or a falsey value". Since the guard clause will make it return `nil`.

I can submit a PR if you want. Just let me know which option you like better. :)

### PR DESCRIPTION
**Pull request by aried3r**
Fixes #20. Finally got around to it. But it changes the expected behavior a bit, according to the tests. So I'm not sure if this is actually a breaking change or "just" a bug fix.

### CODE REVIEW

File: spec/discard/model_spec.rb [hunk 1]
```diff
@@ -81,9 +81,7 @@
 
       describe '#undiscard!' do
         it "raises Discard::RecordNotUndiscarded" do
-          expect {
```
Comment (skatkov) [inline spec/discard/model_spec.rb]: Sorry, I don't follow this. 

We don't expect Discard::RecordNotUndiscarded error anymore? Not sure why is that?

**Comment by gssbzn** [comment]
@aried3r as the one responsible for the bang(!) methods I can tell you that this change is breaking the current functionality and the test was working as expected, additionally returning `true` is a backward braking change but I would let @jhawthorn decide if he decides to introduces that change

**Comment by aried3r** [comment]
Hey @gssbzn, I know it's a breaking change, but either the documentation is wrong (easy for documentation to get out of date) or the non-bang methods might have unexpected behavior for some people. At least within the context of Rails where bang-methods (often) throw exceptions while the ones without bang do not, see [their docs](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html). Although this by no means is a rule.
I think the non-bang methods should, or at least could, be idempotent. Always having to `rescue` because you might be soft-deleting twice is a bit of a nuisance, at least for me. :)

I see an argument for both sides that `Discard::RecordNotDiscarded` might mean _"Sorry, something went wrong, we couldn't discard this record"_ or _"Sorry, something went wrong **or** this record is already discarded, so we couldn't discard it **again**"_

I don't feel strongly about this either way, I'd also be ok with just correcting the documentation, but personally as I mentioned above, having a method to discard without having to catch exceptions would be a nice thing to have, what do you think?

**Comment by aried3r** [comment]
@gssbzn, @jhawthorn, friendly ping. :)

**Comment by gssbzn** [comment]
@aried3r I don't know if this project is still being maintained, no new release has been done with any of the latest changes, I'm ok with making this gem always return `true` when there's no need for a change but I do thing the test needs to be fixed to actually test it raises an error
This would mean that you need to define a `before_discard` callback that aborts the operation, this would be the only way to trigger any error for `(un)discard!`

**Comment by jarednorman** [comment]
Closing in favour of #65.

**Pull request closed without merging**